src/main/java/com/firstspringmvcapp/util/PersonValidator.java

```diff
@@ -24,10 +24,16 @@ public boolean supports(Class<?> aClass) {
 
     @Override
     public void validate(Object o, Errors errors) {
-        Person person = (Person) o;
+        Person person = null;
+        try {
+            person = (Person) o;
+        } catch (Exception e) {
+            e.printStackTrace();
+        }
 
-        if (peopleService.findByName(person.getName()) != null) {
-            errors.rejectValue("name", "", "This name is already taken!");
+        if (peopleService.findByEmail(person.getEmail()) != null) {
```

Comment:
Use could make peopleService.findByEmail return Optional but not null 
var person = peopleService.findByEmail(email); 
if(person.isPresent()) {
}